packages/SwingSet/docs/vat-upgrade.md

```diff
@@ -43,30 +43,36 @@ As vats run, they send and receive messages. When these messages carry object re
 
-There are two basic kinds of exports:
+There are three basic categories of exports (cf. [A Taxonomy of Exo-making Functions](https://github.com/endojs/endo/blob/master/packages/exo/docs/exo-taxonomy.md#heap-vs-virtual-vs-durable)):
 
-* singleton in-RAM objects (created with `Far()` from a standard JavaScript
-  `Object`, using the standard objects-as-closures pattern)
+* Heap objects in vat RAM
+  * one-off objects created with `Far()` or `makeExo()`
+  * instances created with a "make" function from `defineExoClass()`
+  * multifaceted kit instances created with a "makeKit" function from `defineExoClassKit()`
 
-* virtual objects (defined with `makeKind()` or `makeDurableKind()`, created
-  by invoking the function those return, and whose state is kept in the
-  database rather than in RAM)
+* Virtual objects in disk-based storage
+  * instances created with a "make" function from `defineVirtualExoClass()`
+  * multifaceted kit instances created with a "makeKit" function from `defineVirtualExoClassKit()`
 
-The v2 code is obligated to re-define all durable virtual-object Kinds created by v1 during the upgrade phase. Once complete, this allows liveslots to satisfy deserialization of any inbound message addressed to (or referencing) a previously-exported virtual object.
+* Durable objects in disk-based storage
+  * instances created with a "make" function from `defineDurableExoClass()` or `prepareExoClass()`
+  * multifaceted kit instances created with a "makeKit" function from `defineDurableExoClassKit()` or `prepareExoClassKit()`
+  * singleton objects created with `prepareExo()`
 
-The v2 code is also obligated to reattach any singleton objects. The API for this is still under consideration. The root object may be a special case: if the upgrade invocation is simply a new call to `buildRootObject()`, then whatever object that returns will be associated with the root-object vref (`o+0`).
+During the upgrade phase, v2 code is obligated to re-define all durable Kinds created by v1 (i.e., those associated with objects in the third category). Once complete, this allows liveslots to satisfy deserialization of any inbound message addressed to (or referencing) a previously-exported durable object.
 
-## Durable State
+As a special case, the root object returned from v2's `buildRootObject()` is automatically associated with exportID `o+0` (see [How Liveslots Uses the Vatstore](https://github.com/Agoric/agoric-sdk/blob/master/packages/swingset-liveslots/src/vatstore-usage.md#counters)).
```

Comment:
I think this implies that all every result of `buildRootObject()` will have the same identity in liveslots. This wasn't obvious from reading here and would be worth making explicit. Maybe even more explicit: [the root object should not be stored in baggage](https://github.com/Agoric/agoric-sdk/pull/6903#discussion_r1096806998).